acme/acme/challenges.py

```diff
@@ -25,10 +30,10 @@
 class Challenge(jose.TypedJSONObjectWithFields):
     # _fields_to_partial_json
     """ACME challenge."""
-    TYPES: dict = {}
+    TYPES: Dict[str, Type['Challenge']] = {}
 
     @classmethod
-    def from_json(cls, jobj):
+    def from_json(cls, jobj: Dict[str, Any]) -> 'Challenge':
```

Comment:
My opinion is a the well-known advice in Java to always prefer to reference the interface `List` instead of its actual implementations, we should do the same as well here and refer `Mapping` when no specific stuff from `Dict` is required.

Reply:
So I tried to use it except when Mypy was not happy, but given the number of annotations added here, it is more likely that I missed some of them. Thanks a lot!